subprojects/core/src/test/groovy/org/gradle/api/file/FileCollectionSymlinkTest.groovy

```diff
@@ -15,24 +15,34 @@
  */
 package org.gradle.api.file
 
-import org.gradle.testfixtures.ProjectBuilder
-import org.gradle.api.Project
-
-import spock.lang.Specification
-import spock.lang.Shared
+import org.gradle.test.fixtures.AbstractProjectBuilderSpec
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
 import spock.lang.Unroll
 
-// cannot use org.gradle.util.Resources here because the files it returns have already been canonicalized
-class FileCollectionSymlinkTest extends Specification {
-    @Shared Project project = new ProjectBuilder().build()
+@Requires(TestPrecondition.SYMLINKS)
+class FileCollectionSymlinkTest extends AbstractProjectBuilderSpec {
+
+    TestFile baseDir
+    TestFile file
+    TestFile symlink
+    TestFile symlinked
+
+    def setup() {
+        baseDir = temporaryFolder.file('files')
+        file = baseDir.file('file')
+        file.text = 'regular file'
+        symlinked = baseDir.file('symlinked')
+        symlinked.text = 'target of symlink'
+        symlink = baseDir.file('symlink')
+        symlink.createLink(symlinked)
+    }
 
-    @Shared File baseDir = getTestFile("symlinks")
-    @Shared File file = new File(baseDir, "file")
-    @Shared File symlink = new File(baseDir, "symlink")
-    @Shared File symlinked = new File(baseDir, "symlinked")
+    @Unroll
+    def "project.#method can handle symlinks"() {
+        def fileCollection = this."${method}"()
```

Comment:
I think adding `baseDir` as a parameter here would clarify why we have some files in this file-collection.

Reply:
I did that. I am not sure it is better now. WDYT?

Reply:
What I meant is to change this:

    def fileCollection = this."${method}"()

to this:

    def fileCollection = this."${method}"(baseDir)

Reply:
Yeah, but then `files` will have an used parameter.

Reply:
Ah, I see what you did there... I missed that you were calling the method on `this` and not on `project`... I think this is unnecessary magic, and would prefer the previous solution where the code to execute was clearly expressed in the `where:` clause.

Reply:
Better now?